STYLE_GUIDE.adoc

```diff
@@ -3,28 +3,38 @@
 
 toc::[]
 
-= File naming conventions
+= Default conventions
 
-[source,typescript]
-----
-// bad
-import Footer from './Footer/index';
-// good
-import Footer from './Footer';
-----
+By default, we try to follow community conventions. link:https://github.com/basarat/typescript-book/blob/master/docs/styleguide/styleguide.md[Basarat's Style Guide] is used as a reference, with a few exceptions that are listed below:
+
+== File names
 
+Most files are named after the main type / class / interface it's exporting. For instance, `ServiceList.ts` is named after the `ServiceList` interface that it's exporting. Which means that, unlike in Basarat's guide, we often use `PascalCase`.
 
-= Methods
+Files with more general purpose, or the ones that don't export types or such, are `camelCase` (ex: `routes.ts`).
 
-=== Naming
+== Variables, functions, constants
 
-* action types - use constants, [NOUN]_[VERB], eg. USER_FETCH or MATERIAL_REQUEST_UPDATE
-* action creators - [verb][Noun]() eg. fetchCurrentUser(), acceptMaterialRequest()
-* selectors - get[NOUN]() or select[NOUN]() eg. getSelectedCards(), selectActiveRows()
+We mostly follow Basarat's guide, so `camelCase`. Exceptions are:
 
-=== Handlers
+- Redux actions, which are technically constants, are `PascalCase` (ex: `GraphActions`).
+- We sometimes make a difference between constants of global scopes versus constants of local scopes. You may find the former written in `UPPER_SNAKE_CASE` (ex: `const TIMER_REQUEST_PER_SECOND_MIN = 0;`), whereas Basarat's guide makes no distinction between different kinds of constants, all of them being `camelCase`.
```

Comment:
Note here: we should have a clear guideline here instead of the unclear "sometimes". Which means that we should update all the code not following whatever guideline we choose.

What you folk would you prefer: upper snake case on top-level constants, or no scope distinction hence camel case?

Reply:
(Personally I think I prefer to make a distinction and keep upper snake case. It makes it clearer as we read it that it's something top-level defined)

Reply:
Hmmm... IMO, we should use upper snake case for the constants that can be "mapped" as constants in other languages (the `TIMER_REQUEST_PER_SECOND_MIN` is a good example).

For all other constants, we should apply the _variables_ convention (or any other that may could be applied given the case). For example, this one https://github.com/kiali/kiali-ui/blob/387e5b21199c83394eae4489f80661b2edf5cc26/src/components/MetricsOptions/MetricsDuration.tsx#L21 declared as constant, but other languages won't even allow to declare constants in that scope... So, it's ok to apply the convention for variables.

Reply:
@israel-hdez yes, we could do that way, I tried to formalize that analogy with "constants" as understood in other languages by this notion of "top-level constants" versus "local-scoped constants". But our current code is still very far from adopting this pattern.
Just picking a random example, but there's many other examples:
https://github.com/kiali/kiali-ui/blob/master/src/components/CytoscapeGraph/CytoscapeToolbar.tsx#L34-L43

Where we can see side by side top-level camelCase constants and upper-case constants.
So, at the moment we don't really have any rule about that, it's up to the developer.

Reply:
Hmmm... true...
Being strict, the `cytoscapeToolbarStyle` variable cannot be mapped to a `const` in other languages, because compilers need to resolve a static value, while the `cytoscapeToolbarStyle` value can only be resolved at runtime (because it's using a function).

But I can see that this can be very confusing. Not sure how we could remove the "sometimes" subjectivity.

Reply:
depends what is the "other language" :-)
In c/c++ it's pre-processed so you're right.
In java what 'maps' to constants ("static final"), it's still resolved at runtime, allowing function calls, so it would work.
Here there's no preprocessing so no need to consider differently what can only be expressed at runtime or not.

Reply:
Hmm... Yes, I was probably thinking about how C/C++ work... I feel old :'(